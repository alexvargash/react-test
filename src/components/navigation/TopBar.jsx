import styled from 'styled-components';
import Menu from 'components/svgs/Menu';
import Moon from 'components/svgs/Moon';
import Avatar from 'components/users/Avatar';
import InputText from 'components/inputs/InputText';

const Header = styled.header`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 12px 25px;
    background-color: #1E293B;
`;

const FlexWrapper = styled.div`
    display: flex;
    align-items: center;

    & > * + * {
        --tw-space-x-reverse: 0;
        margin-right: calc(1.25rem * var(--tw-space-x-reverse));
        margin-left: calc(1.25rem * calc(1 - var(--tw-space-x-reverse)));
    }
`;

function TopBar(props) {
    return (
        <Header>
            <FlexWrapper>
                <Menu height="25" width="25" />
                <InputText id="search" name="search" onChange={props.onChange} />
            </FlexWrapper>
            <FlexWrapper>
                <Moon height="25" width="25" />
                <Avatar />
            </FlexWrapper>
        </Header>
    );
}

export default TopBar;
