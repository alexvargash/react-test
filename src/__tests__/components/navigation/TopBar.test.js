import { render } from '@testing-library/react';
import TopBar from 'components/navigation/TopBar';

describe('Top Bar', () => {
    it('renders correctly', () => {
        const tree = render(<TopBar />);

        expect(tree).toMatchSnapshot();
    });
});
