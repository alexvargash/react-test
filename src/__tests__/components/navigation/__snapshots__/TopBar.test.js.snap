// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Bar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-hKwCoD gidtkw"
      >
        <div
          class="sc-eCImvq jqHQAg"
        >
          <svg
            class="sc-bdvvaa cffuoN"
            fill="none"
            height="25"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <input
            class="sc-dkPtyc pCpHc"
            id="search"
            name="search"
            type="text"
          />
        </div>
        <div
          class="sc-eCImvq jqHQAg"
        >
          <svg
            class="sc-bdvvaa cffuoN"
            fill="none"
            height="25"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <div
            class="sc-gsDJrp kTMNQz"
          >
            <svg
              class="sc-bdvvaa cVEzbr"
              color="#1E293B"
              fill="none"
              height="20"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-hKwCoD gidtkw"
    >
      <div
        class="sc-eCImvq jqHQAg"
      >
        <svg
          class="sc-bdvvaa cffuoN"
          fill="none"
          height="25"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <input
          class="sc-dkPtyc pCpHc"
          id="search"
          name="search"
          type="text"
        />
      </div>
      <div
        class="sc-eCImvq jqHQAg"
      >
        <svg
          class="sc-bdvvaa cffuoN"
          fill="none"
          height="25"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <div
          class="sc-gsDJrp kTMNQz"
        >
          <svg
            class="sc-bdvvaa cVEzbr"
            color="#1E293B"
            fill="none"
            height="20"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
